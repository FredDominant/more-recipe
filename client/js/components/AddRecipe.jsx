import React from 'react';
import { connect } from 'react-redux';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
/**
 * @description componet holds form to add recipes
 *
 * @class AddRecipe
 * @extends {React.Component}
 */
class AddRecipe extends React.Component {
/**
 * Creates an instance of AddRecipe.
 * @param {any} props
 * @memberof AddRecipe
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      recipeImage: '',
      errors: {}
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof AddRecipe
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onUpload(event) {
    const image = event.target.files[0];
    this.setState({ recipeImage: image });
  }
  /**
 *
 * @returns {null} null
 * @memberof AddRecipe
 */
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
   *
   * @returns {null} null
   * @memberof AddRecipe
   */
  addRecipe() {
    uploadImage(this.state.recipeImage)
      .then((response) => {
        // console.log('image url after axios is', response.data.secure_url);
        this.setState({ recipeImage: response.data.secure_url });
        // console.log('image url after updating state is', this.state.recipeImage);
        // dispatch this to store;
      })
      .catch((error) => {
        console.log(error);
      });
  }
  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addRecipe();
    }
  }
  /**
   *
   * @returns {HTML} html
   * @memberof AddRecipe
   */
  render() {
    const errors = this.state.errors;
    return (
      <div>
        <div className="modal fade" id="addRecipe" tabIndex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
          <div className="modal-dialog" role="document">
            <div className="modal-content container">
              <div className="modal-header">
                <h5 className="modal-title"><span id="title"><h3 title="More Recipes and cooking tips">More Recipes</h3></span></h5>
                <button type="button" className="close" data-dismiss="modal" aria-label="Close">
                  <span aria-hidden="true">&times;</span>
                </button>
              </div>
              <div className="modal-body">
                <div className="container error-body">
                  {errors.name && <div className="alert alert-dismissible alert-danger" role="alert">{errors.name}</div>}
                  {errors.description && <div className="alert alert-dismissible alert-danger" role="alert">{errors.description}</div>}
                  {errors.ingredients && <div className="alert alert-dismissible alert-danger" role="alert">{errors.ingredients}</div>}
                  {errors.directions && <div className="alert alert-dismissible alert-danger" role="alert">{errors.directions}</div>}
                </div>
                <form onSubmit={this.handleSubmit}>
                  <div className="form-group">
                    <label htmlFor="exampleInputEmail1">Recipe Name</label>
                    <input
                      type="text"
                      className="form-control"
                      name="name"
                      id="recipeName"
                      aria-describedby="small-recipe-name"
                      placeholder="Awesome Recipe!"
                      value={this.state.name}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipe-description">Recipe Description:</label>
                    <input
                      type="text"
                      className="form-control"
                      name="description"
                      id="recipe-description"
                      placeholder="Awesome sauce made out of nothing!"
                      value={this.state.description}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Ingredients:</label>
                    <textarea
                      className="form-control"
                      id="cook-directions"
                      name="ingredients"
                      rows="3"
                      placeholder="Onions, tomatoes, grilled chicken, nutmeg"
                      value={this.state.ingredients}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="cook-directions">Directions to cook:</label>
                    <textarea
                      className="form-control"
                      name="directions"
                      id="cook-directionss"
                      rows="3"
                      placeholder="dice tomatoes, tear chicken, boil onions, mix with nutmeg"
                      value={this.state.directions}
                      onChange={this.onChange}
                    />
                  </div>

                  <div className="form-group">
                    <label htmlFor="recipeImage">Recipe Image</label>
                    <input
                      type="file"
                      className="form-control-file"
                      name="file"
                      id="recipeImage"
                      onChange={this.onUpload}
                    />
                  </div>
                  <button className="btn btn-primary">Add Recipe</button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({

});
export default connect(mapStateToProps)(AddRecipe);
