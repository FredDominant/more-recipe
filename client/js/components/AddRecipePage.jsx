import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import MDSpinner from 'react-md-spinner';
import toastr from 'toastr';

import uploadImage from '../utils/uploadImage';
import recipeValidator from '../validation/recipeValidator';
import addRecipe from '../actions/addRecipe';
/**
 *
 * @class AddRecipePage
 *
 * @extends {React.Component}
 */
export class AddRecipePage extends React.Component {
/**
 * @description Creates an instance of AddRecipePage.
 *
 * @param {any} props
 *
 * @memberof AddRecipePage
 */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      description: '',
      ingredients: '',
      directions: '',
      isUploading: false,
      recipeImage: '/images/noImage.png',
      errors: {},
      addRecipeError: '',
      uploadImageError: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
    this.handleSelectImage = this.handleSelectImage.bind(this);
  }
  /**
 * @description component lifecycle method
 *
 * @returns {null} null
 *
 * @param {nextProp} nextProp object from store
 *
 * @memberof AddRecipe
 */
  componentWillReceiveProps(nextProp) {
    this.setState({ addRecipeError: nextProp.errorMessage });
  }

  /**
 *
 * @returns {null} null
 * @param {any} event
 * @memberof AddRecipe
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  onUpload(event) {
    const reader = new FileReader();
    reader.onload = (e) => {
      this.setState({ recipeImage: e.target.result });
    };
    reader.readAsDataURL(event.target.files[0]);
  }
  /**
*
* @returns {null} null

* @memberof AddRecipe
*/
  isValid() {
    const { errors, isValid } = recipeValidator(this.state);
    if (!isValid) {
      this.setState({ errors });
    }
    return isValid;
  }
  /**
 *
 * @returns {null} null
 *
 * @memberof AddRecipe
 */
  addNewRecipe() {
    const { recipeImage, name, description, directions, ingredients } = this.state;
    if (recipeImage !== '/images/noImage.png') {
      this.setState({ isUploading: true });
      return uploadImage(recipeImage)
        .then((response) => {
          const recipeUrl = response.data.secure_url;
          console.log('recipe url is', recipeUrl);
          this.setState({ recipeImage: recipeUrl, isUploading: false });
          this.props.createRecipe(this.state);
        })
        .catch(() => {
          this.setState({ isUploading: false, uploadImageError: 'Could not upload image. Recipe not added' });
          toastr.options = {
            closeButton: true
          };
          toastr.error(this.state.uploadImageError);
        });
    }
    console.log('jumped to this place');
    this.props.createRecipe({ name, directions, ingredients, description });
  }
  /**
*
* @returns {null} null

* @param {any} event

* @memberof AddRecipe
*/
  handleSubmit(event) {
    event.preventDefault();
    if (this.isValid()) {
      this.addNewRecipe();
    }
  }
  /**
   *
   * @memberof AddRecipePage
   *
   * @returns {null} null
   */
  handleSelectImage() {
    $('input[type=file]').click();
  }
  /**
   *
   * @returns {null} null
   *
   * @memberof AddRecipePage
   */
  render() {
    const { errors, recipeImage, name, description, directions, ingredients } = this.state;
    return (
      <div className="container-fluid">
        <div id="add-recipe-form" className="container">
          <form onSubmit={this.handleSubmit}>
            <div className="row">
              <div className="col-sm-4">
                <div
                  className="recipe-image"
                  id="add-recipe-image-container"
                  onClick={this.handleSelectImage}
                  role="presentation"
                >
                  <img
                    className="img-thumbnail"
                    src={recipeImage}
                    alt=""
                    srcSet=""
                    id="add-recipe-image"
                  />
                </div>
                <br />
                <input
                  style={{ display: 'none' }}
                  type="file"
                  name="file"
                  id="file-upload"
                  onChange={this.onUpload}
                />
              </div>
              <div className="col-sm-8">
                <div className="form-group">
                  <label htmlFor="recipeName">Recipe Name</label>
                  <input
                    type="text"
                    id="recipeName"
                    className="form-control"
                    name="name"
                    value={name}
                    onChange={this.onChange}
                  />
                  {errors.name && <small className="form-text text-muted">
                    <span className="error-text"> {errors.name} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDescription">Recipe Description</label>
                  <input
                    type="text"
                    id="recipeDescription"
                    className="form-control"
                    name="description"
                    value={description}
                    onChange={this.onChange}
                  />
                  {errors.description && <small className="form-text text-muted">
                    <span className="error-text"> {errors.description} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeDirections">Recipe Directions</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeDirections"
                    className="form-control"
                    name="directions"
                    value={directions}
                    onChange={this.onChange}
                  />
                  {errors.directions && <small className="form-text text-muted">
                    <span className="error-text"> {errors.directions} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <label htmlFor="recipeingredients">Ingredients</label>
                  <textarea
                    type="text"
                    rows="5"
                    id="recipeIngredients"
                    className="form-control"
                    placeholder="Comma separated list of ingredients"
                    name="ingredients"
                    value={ingredients}
                    onChange={this.onChange}
                  />
                  {errors.ingredients && <small className="form-text text-muted">
                    <span className="error-text"> {errors.ingredients} </span>
                  </small>}
                </div>

                <div className="form-group">
                  <button className="btn btn-primary">
                    {
                      !this.state.isUploading &&
                      <h6 className="text-center"> Add Recipe </h6>
                    }
                    {
                      this.state.isUploading &&
                      <h6 className="text-center"> Uploading Image... </h6>
                    }
                    {
                      this.props.loading &&
                      <span> <MDSpinner /></span>
                    }
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  errorMessage: state.addRecipe.addRecipeErrorMessage,
  addRecipeSuccess: state.addRecipe.addRecipeSuccess,
  loading: state.isUploading
});

const mapDispatchToProps = dispatch => ({
  createRecipe: recipe => dispatch(addRecipe(recipe))
});

AddRecipePage.propTypes = {
  createRecipe: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired
};
AddRecipePage.defaultProps = {
  errorMessage: null,
  addRecipeSuccess: null
};
export default connect(mapStateToProps, mapDispatchToProps)(AddRecipePage);

