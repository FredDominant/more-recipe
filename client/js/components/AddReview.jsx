import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import addReview from '../actions/addReview';
/**
 *
 * @class AddReview
 *
 * @extends {React.Component}
 */
export class AddReview extends React.Component {
/**
 * @description Creates an instance of AddReview.
 *
 * @param {object} props
 *
 * @memberof AddReview
 */
  constructor(props) {
    super(props);
    this.state = {
      content: ''
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  /**
   *
   * @param {object} event
   *
   * @memberof AddReview
   *
   * @returns {null} null
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
   * @returns {null} null
   *
   * @param {object} event
   *
   * @memberof AddReview
   */
  handleSubmit(event) {
    const { content } = this.state;
    const recipeId = this.props.recipeId;
    event.preventDefault();
    if (content.trim().length) {
      this.props.postReview(content, recipeId);
      this.setState({ content: '' });
    }
  }
  /**
   *
   *
   * @memberof AddReview
   *
   * @return {ReactElement} markup
   */
  render() {
    return (
      <div>
        <div id="review-form">
          <div>
            <div>
              <form onSubmit={this.handleSubmit}>
                <input
                  type="text"
                  className="form-control"
                  name="content"
                  placeholder="Add a review..."
                  onChange={this.onChange}
                  value={this.state.content}
                  required
                />
                <br />
                <button className="btn" id="add-review-button">
                  <span><i className="fas fa-comment-alt" /> </span>
                  Post Review
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
AddReview.propTypes = {
  postReview: PropTypes.func.isRequired,
  recipeId: PropTypes.number
};
AddReview.defaultProps = {
  recipeId: 0
};
const mapDispatchToProps = dispatch => ({
  postReview: (review, recipeId) => dispatch(addReview(review, recipeId))
});
export default connect(null, mapDispatchToProps)(AddReview);

