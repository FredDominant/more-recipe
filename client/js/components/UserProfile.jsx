import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import NavBar from './Navbar';
import userProfile from '../actions/userProfile';
import updateProfile from '../actions/editProfile';
import uploadImage from '../utils/uploadImage';
/**
 *
 *
 * @class UserProfile
 * @extends {React.Component}
 */
class UserProfile extends React.Component {
/**
 * Creates an instance of UserProfile.
 * @param {any} props
 * @memberof UserProfile
 */
  constructor(props) {
    super(props);
    this.state = {
      firstname: '',
      lastname: '',
      email: '',
      password: '',
      picture: '',
      disabled: true,
      uploadImageError: '',
      selectedImage: false
    };
    this.onChange = this.onChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.onUpload = this.onUpload.bind(this);
    this.onEdit = this.onEdit.bind(this);
  }
  /**
   *
   * @returns {null} null
   * @memberof UserProfile
   */
  componentDidMount() {
    this.props.viewProfile();
  }
  /**
   *
   * @returns {null} null
   * @param {any} nextProps
   * @memberof UserProfile
   */
  componentWillReceiveProps(nextProps) {
    console.log('next Props is', nextProps);
    const { firstname, lastname, email, picture } = nextProps.userDetails;
    this.setState({ firstname, lastname, email, picture });
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof UserProfile
   */
  onUpload(event) {
    const reader = new FileReader();
    reader.onload = (e) => {
      this.setState({ picture: e.target.result });
    };
    reader.readAsDataURL(event.target.files[0]);
    this.setState({ selectedImage: true });
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof UserProfile
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof UserProfile
   */
  onEdit(event) {
    event.preventDefault();
    this.setState({ disabled: !this.state.disabled });
  }
  /**
   *
   * @returns {null} null
   * @param {any} event
   * @memberof UserProfile
   */
  handleSubmit(event) {
    const { firstname, lastname, email, picture } = this.state;
    event.preventDefault();
    if (this.state.selectedImage) {
      return uploadImage(picture)
        .then((response) => {
          const url = response.data.secure_url;
          this.setState({ picture: url });
          this.props.updateProfile(this.state);
        })
        .catch((error) => {
          this.setState({ uploadImageError: error.error.message });
        });
    }
    this.props.updateProfile({ firstname, lastname, email, picture });
  }

  /**
   *
   * @returns {null} null
   * @memberof UserProfile
   */
  render() {
    return (
      <div >
        <NavBar />
        {this.props.updateSuccess && <div className="container">
          <div className="alert alert-success alert-dismissible" role="alert">Profile updated</div>
        </div>}
        <div className="container" id="update-profile-form">
          <div id="update-profile-body">
            <form onSubmit={this.handleSubmit}>
              {/* <div className="row"> */}
              <div className="">
                <div id="user-image-container">
                  <div className="profile-image">
                    <img className="img-thumbnail" id="update-profile-picture" src={this.state.picture} alt={this.state.firstname} srcSet="" />
                  </div>
                  <br />
                  <input
                    type="file"
                    name="file"
                    id="profile-upload"
                    onChange={this.onUpload}
                    disabled={this.state.disabled}
                  />
                </div>
                <br />

                <div className="form-group">
                  <label htmlFor="profile-firstname">First Name</label>
                  <input
                    type="text"
                    id="profile-firstname"
                    className="form-control"
                    name="firstname"
                    value={this.state.firstname}
                    onChange={this.onChange}
                    disabled={this.state.disabled}
                  />
                </div>

                <div className="form-group">
                  <label htmlFor="profile-lastname">Last Name</label>
                  <input
                    type="text"
                    id="profile-lastname"
                    className="form-control"
                    name="lastname"
                    value={this.state.lastname}
                    onChange={this.onChange}
                    disabled={this.state.disabled}
                  />
                </div>

                <div className="form-group">
                  <label htmlFor="profile-email">Email</label>
                  <input
                    type="email"
                    id="profile-email"
                    className="form-control"
                    name="email"
                    value={this.state.email}
                    onChange={this.onChange}
                    disabled={this.state.disabled}
                  />
                </div>

                <div className="form-group">
                  <label htmlFor="profile-password">Password</label>
                  <input
                    type="text"
                    id="profile-password"
                    className="form-control"
                    name="password"
                    value={this.state.password}
                    onChange={this.onChange}
                    disabled={this.state.disabled}
                  />
                </div>

                <div className="form-group">
                  <div className="row">
                    <div className="col-sm-2">
                      <button className="btn btn-success" disabled={this.state.disabled}>Update</button></div>
                    <div className="col-sm-1" />
                    <div className="col-sm-2">
                      <button className="btn btn-primary" onClick={this.onEdit}>Edit</button>
                    </div>
                  </div>
                </div>
              </div>
              {/* </div> */}
            </form>
          </div>
        </div>
      </div>
    );
  }
}
const mapDispatchToprops = dispatch => ({
  viewProfile: () => dispatch(userProfile()),
  updateProfile: updateDetails => dispatch(updateProfile(updateDetails))
});

const mapStateToProps = state => ({
  userDetails: state.currentUserProfile.User,
  updateSuccess: state.currentUserProfile.UpdateSuccess
});

UserProfile.propTypes = {
  viewProfile: PropTypes.func.isRequired,
  userDetails: PropTypes.shape(),
  updateProfile: PropTypes.func.isRequired,
  updateSuccess: PropTypes.bool
};
UserProfile.defaultProps = {
  userDetails: {},
  updateSuccess: false
};

export default connect(mapStateToProps, mapDispatchToprops)(UserProfile);

