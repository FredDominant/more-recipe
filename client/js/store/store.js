const store = {
  loginForm: {
    email: '',
    password: ''
  },
  signunForm: {
    firstname: '',
    lastname: '',
    email: '',
    password: '',
    confirmPassword: ''
  }
};
export default store;
