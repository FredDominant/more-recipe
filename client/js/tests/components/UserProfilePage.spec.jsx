import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';

import store from '../../store/store';
import { UserProfile } from '../../components/UserProfile';
import mockData from '../mocks/recipes.mock';

const props = {
  viewProfile: jest.fn(),
  userDetails: {},
  updateProfile: jest.fn(() => Promise.resolve()),
};

const state = {
  firstName: '',
  lastName: '',
  email: '',
  password: '',
  confirmPassword: '',
  imageUrl: '',
  errors: {},
  disabled: true,
  uploadImageError: '',
  selectedImage: false,
  uploading: false,
};
/**
 *
 *
 * @returns {null} null
 */
const setup = () => {
  const shallowWrapper = shallow(<UserProfile {...props} />);
  const mountedWrapper = mount(
    <Provider store={store}>
      <BrowserRouter>
        <UserProfile {...props} store={store} />
      </BrowserRouter>
    </Provider>
  );

  return {
    shallowWrapper,
    mountedWrapper,
    props
  };
};

describe('UserProfile component should', () => {
  sinon.spy(UserProfile.prototype, 'componentWillReceiveProps');
  it('render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('ComponentWillRecieveProps()', () => {
  it('should be called when component is passed in new props', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    wrapper.setState(state);
    wrapper.instance().componentWillReceiveProps(props);
    expect(UserProfile.prototype.componentWillReceiveProps.calledOnce).toEqual(true);
  });
});


describe('onChange', () => {
  it('should set form value', () => {
    const event = {
      target: { name: 'firstName', value: 'new name' }
    };
    const { shallowWrapper } = setup();
    const password = shallowWrapper.find('#profile-firstName');
    password.simulate('change', event);
    expect(shallowWrapper.instance().state.firstName).toBe('new name');
  });
});

describe('onUpload()', () => {
  beforeEach(() => {
    global.FileReader = () => ({
      readAsDataURL: () => {},
      onload: e => e
    });
    global.reader = {
      onload: () => {}
    };
  });

  it('should select an image when clicked', () => {
    const { shallowWrapper } = setup();
    const event = mockData.uploadImage;
    const action = shallowWrapper.instance();
    action.onUpload(event);
    action.setState({
      imageUrl: event.target.files
    });
    expect(action.state.imageUrl).toBe(event.target.files);
  });
});

describe('onEdit', () => {
  it('should set form value', () => {
    const event = {
      preventDefault: jest.fn()
    };
    const shallowWrapper = shallow(<UserProfile {...props} />);
    const password = shallowWrapper.find('#edit');
    password.simulate('click', event);
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('onSelectImage()', () => {
  it('should be called when image is clicked', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    wrapper.setState(state);
    wrapper.instance().onSelectImage();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('handleSubmit()', () => {
  it('should not work if passwords do not match', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    sinon.spy(UserProfile.prototype, 'handleSubmit');
    const event = {
      preventDefault: jest.fn()
    };
    const form = wrapper.find('form');
    wrapper.setState({
      firstName: 'firstName',
      lastName: 'lastName',
      email: 'email@email.com',
      password: '123456',
      confirmPassword: '1234567',
    });
    form.simulate('submit', event);
    expect(wrapper.instance().state.errors).toEqual({
      confirmPassword: "Passwords don't match",
      password: "Passwords don't match",
    });
  });

  it('should work if passwords match', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    const event = {
      preventDefault: jest.fn()
    };
    const form = wrapper.find('form');
    wrapper.setState({
      firstName: 'firstName',
      lastName: 'lastName',
      email: 'email@email.com',
      password: '123456',
      confirmPassword: '123456',
    });
    form.simulate('submit', event);
    expect(wrapper.instance().state.errors).toEqual({});
  });

  it('should work if password is not updated', () => {
    const wrapper = shallow(<UserProfile {...props} />);
    const event = {
      preventDefault: jest.fn()
    };
    const form = wrapper.find('form');
    wrapper.setState({
      firstName: 'firstName',
      lastName: 'lastName',
      email: 'email@email.com',
    });
    form.simulate('submit', event);
    expect(wrapper.instance().state.errors).toEqual({});
  });
});
