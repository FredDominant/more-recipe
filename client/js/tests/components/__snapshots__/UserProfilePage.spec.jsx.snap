// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile component should render correctly 1`] = `
<div
  className="profile-body"
>
  <div
    className="container"
    id="update-profile-form"
  >
    <div
      id="update-profile-body"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className=""
        >
          <div
            id="user-image-container"
          >
            <div
              className="profile-image"
              id="user-profile-image-container"
              onClick={[Function]}
              role="presentation"
            >
              <img
                alt=""
                className="img-thumbnail"
                id="update-profile-image"
                src=""
                srcSet=""
              />
            </div>
            <br />
            <input
              accept=".png,.gif,.jpg,.jpeg"
              disabled={true}
              id="profile-upload"
              name="file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-firstName"
            >
              First Name
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-firstName"
              name="firstName"
              onChange={[Function]}
              pattern="[A-Za-z]+"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-lastName"
            >
              Last Name
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-lastName"
              name="lastName"
              onChange={[Function]}
              pattern="[A-Za-z]+"
              required={true}
              type="text"
              value=""
            />
          </div>
          <br />
          <h6
            className="text-left"
          >
            Account security setting
          </h6>
          <hr />
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-email"
            >
              Email
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-email"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-password"
            >
              Password
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-confirmPassword"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-password"
            >
              Confirm Password
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-password"
              name="confirmPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <div
              className="container row"
            >
              <div
                className="col-sm-2 col-md-2 col-lg-2"
              >
                <button
                  className="btn btn-primary"
                  id="edit"
                  onClick={[Function]}
                >
                  <span>
                    <i
                      className="fas fa-unlock"
                    />
                  </span>
                </button>
              </div>
              <div
                className="col-sm-10 col-md-10 col-lg-10"
              >
                <button
                  className="btn btn-success"
                  disabled={true}
                >
                  update profile
                </button>
              </div>
              <div
                className="col-sm-2 col-md-2 col-lg-2"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <br />
  <Footer />
</div>
`;

exports[`onEdit should set form value 1`] = `
<div
  className="profile-body"
>
  <div
    className="container"
    id="update-profile-form"
  >
    <div
      id="update-profile-body"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className=""
        >
          <div
            id="user-image-container"
          >
            <div
              className="profile-image"
              id="user-profile-image-container"
              onClick={[Function]}
              role="presentation"
            >
              <img
                alt=""
                className="img-thumbnail"
                id="update-profile-image"
                src=""
                srcSet=""
              />
            </div>
            <br />
            <input
              accept=".png,.gif,.jpg,.jpeg"
              disabled={false}
              id="profile-upload"
              name="file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-firstName"
            >
              First Name
            </label>
            <input
              className="form-control"
              disabled={false}
              id="profile-firstName"
              name="firstName"
              onChange={[Function]}
              pattern="[A-Za-z]+"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-lastName"
            >
              Last Name
            </label>
            <input
              className="form-control"
              disabled={false}
              id="profile-lastName"
              name="lastName"
              onChange={[Function]}
              pattern="[A-Za-z]+"
              required={true}
              type="text"
              value=""
            />
          </div>
          <br />
          <h6
            className="text-left"
          >
            Account security setting
          </h6>
          <hr />
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-email"
            >
              Email
            </label>
            <input
              className="form-control"
              disabled={false}
              id="profile-email"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-password"
            >
              Password
            </label>
            <input
              className="form-control"
              disabled={false}
              id="profile-confirmPassword"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-password"
            >
              Confirm Password
            </label>
            <input
              className="form-control"
              disabled={false}
              id="profile-password"
              name="confirmPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <div
              className="container row"
            >
              <div
                className="col-sm-2 col-md-2 col-lg-2"
              >
                <button
                  className="btn btn-primary"
                  id="edit"
                  onClick={[Function]}
                >
                  <span>
                    <i
                      className="fas fa-lock"
                    />
                  </span>
                </button>
              </div>
              <div
                className="col-sm-10 col-md-10 col-lg-10"
              >
                <button
                  className="btn btn-success"
                  disabled={false}
                >
                  update profile
                </button>
              </div>
              <div
                className="col-sm-2 col-md-2 col-lg-2"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <br />
  <Footer />
</div>
`;

exports[`onSelectImage() should be called when image is clicked 1`] = `
<div
  className="profile-body"
>
  <div
    className="container"
    id="update-profile-form"
  >
    <div
      id="update-profile-body"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className=""
        >
          <div
            id="user-image-container"
          >
            <div
              className="profile-image"
              id="user-profile-image-container"
              onClick={[Function]}
              role="presentation"
            >
              <img
                alt=""
                className="img-thumbnail"
                id="update-profile-image"
                src=""
                srcSet=""
              />
            </div>
            <br />
            <input
              accept=".png,.gif,.jpg,.jpeg"
              disabled={true}
              id="profile-upload"
              name="file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
          </div>
          <br />
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-firstName"
            >
              First Name
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-firstName"
              name="firstName"
              onChange={[Function]}
              pattern="[A-Za-z]+"
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-lastName"
            >
              Last Name
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-lastName"
              name="lastName"
              onChange={[Function]}
              pattern="[A-Za-z]+"
              required={true}
              type="text"
              value=""
            />
          </div>
          <br />
          <h6
            className="text-left"
          >
            Account security setting
          </h6>
          <hr />
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-email"
            >
              Email
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-email"
              name="email"
              onChange={[Function]}
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-password"
            >
              Password
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-confirmPassword"
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              htmlFor="profile-password"
            >
              Confirm Password
            </label>
            <input
              className="form-control"
              disabled={true}
              id="profile-password"
              name="confirmPassword"
              onChange={[Function]}
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <div
              className="container row"
            >
              <div
                className="col-sm-2 col-md-2 col-lg-2"
              >
                <button
                  className="btn btn-primary"
                  id="edit"
                  onClick={[Function]}
                >
                  <span>
                    <i
                      className="fas fa-unlock"
                    />
                  </span>
                </button>
              </div>
              <div
                className="col-sm-10 col-md-10 col-lg-10"
              >
                <button
                  className="btn btn-success"
                  disabled={true}
                >
                  update profile
                </button>
              </div>
              <div
                className="col-sm-2 col-md-2 col-lg-2"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
  <br />
  <Footer />
</div>
`;
